scaaml/io/dataset.py

```diff
@@ -40,11 +39,28 @@
 from scaaml.io.spell_check import find_misspellings
 import scaaml.io.utils as siutils
 from scaaml.io.utils import dtype_name_to_dtype, dtype_dtype_to_name
-from scaaml.io.shard import Shard
+from scaaml.io.shard import Shard, CompressionT
 from scaaml.io.errors import DatasetExistsError
 
+_T = TypeVar("_T")
+
+
+class FakeTqdm(Generic[_T]):
```

Comment:
NIT This is not necessary, tqdm has a parameter `disable: bool`

Reply:
That would require more code changes though because the function that builds the progressbar takes a class instead of a bool as a parameter.
I prefer leaving this to another PR, this one only deals with type checking.